.github/workflows/commit_shelter.yml

```diff
@@ -1,55 +1,146 @@
+        apt-get update -y && apt-get install -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-dev libsgx-uae-service;
+        cp -rf /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Install centos dependency
+      if: ${{ contains(matrix.tag, 'centos') }}
       run: |
-        docker exec $rune_test bash -c "cd /root/inclavare-containers-test/shelter;
-        make -j1;
-        ./shelter remoteattestation"
+        docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }};
+        yum -y install yum-utils wget iptables;
+        wget -c https://download.01.org/intel-sgx/sgx-linux/2.13/distro/centos8.2-server/sgx_rpm_local_repo.tgz;
+        tar xzf sgx_rpm_local_repo.tgz;
+        yum-config-manager --add-repo sgx_rpm_local_repo;
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tgz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Install alinux dependency
+      if: ${{ contains(matrix.tag, 'alinux') }}
+      run: docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }};
+        yum -y install yum-utils wget tar procps alinux-release-experimentals  protobuf-c-devel;
+        wget -c https://mirrors.openanolis.org/inclavare-containers/alinux2/sgx_rpm_local_repo.tar.gz;
+        tar xzf sgx_rpm_local_repo.tar.gz;
+        yum-config-manager --add-repo file:/root/inclavare-containers/${{ matrix.tag }}/sgx_rpm_local_repo;
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tar.gz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Run enclave tls server for null quote
+      if: ${{ contains(matrix.sgx, 'SGX1') }}
+      run: docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -a nullquote' &
+
+    - name: Run shelter for null quote
+      if: ${{ contains(matrix.sgx, 'SGX1') }}
+      run: docker exec $inclavare_test bash -c 'ls /root/inclavare-containers/${{ matrix.tag }} && cd /root/inclavare-containers/${{ matrix.tag }}/shelter && ./shelter remoteattestation  --verifier nullquote'
+        docker restart $inclavare_test;
+
+    - name: Run enclave tls server for mutual sgx_ecdsa
+      if: ${{ contains(matrix.sgx, 'SGX2') }}
+      run: |
+        docker cp /etc/sgx_default_qcnl.conf  $inclavare_test:/etc/;
+        docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -l debug -a sgx_ecdsa -i 127.0.0.1 -p 1235 -m' &
```

Comment:
If multiple jobs start `encalve-tls-server` in multiple test images at the same time, then the error of `address already in use` will occur. Therfore, it is better to use a random port instead of 1235 and use `netstat -ntl` to check whether the port is already in use.

Reply:
Good questions for such type parallel running to use the same port on host by "-net host".

Reply:
As we discussing offline, we find the way and i will test the solution in local, after pass the test i will push it into this PR.

Reply:
Why the port is still fixed?

Reply:
Random port code is ongoing.

Reply:
Random port fix is already submitted.

Reply:
Local test by push trigger as below:
https://github.com/zhiminghufighting/inclavare-containers-1/runs/2455905163?check_suite_focus=true